src/v/cluster/tm_stm.cc

```diff
@@ -367,11 +355,11 @@ tm_stm::do_update_tx(tm_transaction tx, model::term_id term) {
         co_return tm_stm::op_status::unknown;
     }
 
-    auto tx_opt = do_get_tx(tx.id);
-    if (!tx_opt.has_value()) {
+    auto tx_it = _log_txes.find(tx.id);
```

Comment:
Why did you get tx only from log_state?

Reply:
we replicate & wait so the batch is applied and when it happens the new tx erases old from the mem and put it to log